oraclizeAPI_0.5.sol

```diff
@@ -1347,7 +1347,9 @@ contract usingOraclize {
         internal
         returns (bytes32 _id)
     {
-        uint256 price = oraclize.getPrice(_datasource);
+        uint256 price = oraclize.addressCustomPaymentToken(address(this)) == address(0)

```

Comment:
I'm almost wondering if we should make another overload ~nightmare~ of oraclize_query(..., supportTokens = true) or oraclize_query_token?

I think these costs will be quite significant that it'll sacrifice optimizations, and push costs onto everyone.

Reply:
Have discussed this elsewhere to resolving the conversation here.

Reply:
Sorry for the confusion on this and telling you to go this route first, I was initially under the impression there was just a ternary inclusion. You did also mention a storage read, which would've still been okay, as that would be 200 gas, and I considered we had 500/600 gas to play with. Of course, it's an external call involving a storage read, which can get quite expensive.

Reply:
No worries, I was too vague in pointing it out initialy. No harm done, will have the changes on your (zenhub) desk by (your timezone's) morning.